### PR TITLE
Jp okta 417473

### PR DESCRIPTION
**Pull request by jaredperreault-okta**

### CODE REVIEW

File: lib/idx/remediators/Base/Remediator.ts [hunk 1]
```diff
@@ -182,6 +182,10 @@ export class Remediator {
       if (Array.isArray(input)) {
         input.forEach(i => inputs.push(i));
       } else {
+        // guarantees field-level messages are passed back
+        if (messages) {
+          input.messages = messages;
```
Comment (jaredperreault-okta) [inline lib/idx/remediators/Base/Remediator.ts:187]: field-level messages fix

File: lib/OktaAuth.ts [hunk 2]
```diff
@@ -336,8 +335,6 @@ class OktaAuth implements OktaAuthInterface, SigninAPI, SignoutAPI {
       unlockAccount: unlockAccount.bind(null, this),
     };
 
-    setGlobalRequestInterceptor(createGlobalRequestInterceptor(this)); // to pass custom headers to IDX endpoints
```
Comment (shuowu) [inline lib/OktaAuth.ts:339]: Nice! this is what we need for the tree shaking next step

File: lib/idx/interact.ts [hunk 3]
```diff
@@ -35,6 +35,20 @@ export interface InteractResponse {
   meta: IdxTransactionMeta;
 }
 
+/* eslint-disable camelcase */
+export interface InteractParams {
```
Comment (shuowu) [inline lib/idx/interact.ts:39]: remove export to keep this type private

File: lib/idx/remediators/Base/Remediator.ts (hunk 1 above)
Comment (shuowu) [reply lib/idx/remediators/Base/Remediator.ts:187]: I remember we also have a jira for this issue, we can close that one together.

File: test/spec/OktaAuth/constructor.ts [hunk 4]
```diff
@@ -24,18 +24,9 @@ jest.mock('../../../lib/features', () => {
...
-    ...jest.requireActual('../../../lib/idx/headers') as any,
-    setGlobalRequestInterceptor: jest.fn()
-  };
-});
-
-const mocked = {
-  features: require('../../../lib/features'),
-  idxHeaders: require('../../../lib/idx/headers')
-};
+// const mocked = {
```
Comment (shuowu) [inline test/spec/OktaAuth/constructor.ts:27]: remove?

File: test/spec/idx/idxState/unit/v1/generateIdxAction.test.js [hunk 5]
```diff
@@ -11,50 +11,60 @@
...
 
-jest.mock('cross-fetch');
 /*
   Doing a jest.mock('../../src/makeIdxState') has problems with jest.mock causing the test to hang
   and spikes up the CPU usage for the current node process.
   Alternative mocking approach: https://jestjs.io/docs/en/es6-class-mocks
 */
 const mockMakeIdxState = jest.fn();
-jest.mock('../../../../../../lib/idx/idx-js/v1/makeIdxState', () => ({
+jest.mock('../../../../../../lib/idx/idxState/v1/makeIdxState', () => ({
```
Comment (shuowu) [inline test/spec/idx/idxState/unit/v1/generateIdxAction.test.js:43]: maybe in a future refactor we can map `../../../../../../lib/` to an alias in jest config with `moduleNameMapper `

**Review by shuowu — APPROVED** [review]
Looks great! Please also add changelogs

File: lib/idx/interact.ts (hunk 3 above)
Comment (jaredperreault-okta) [reply lib/idx/interact.ts:39]: this type is used in a few other places: https://github.com/okta/okta-auth-js/blob/jp-OKTA-417473/lib/types/api.ts#L298

**Review by aarongranick-okta — APPROVED** [review]

**Pull request closed without merging**